spec/acceptance/etcd_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'spec_helper_acceptance'
+
+describe 'golang class:' do
+  context 'default parameters' do
+    it 'runs successfully' do
+      pp = <<-EOS
+      class { 'etcd':
```

Comment:
should probably start with a simple `include etcd`

Reply:
not possible because we rely `data-dir` to be supplied in `$config` parameter. If we moved that to a parameter like `$data_dir` with a default then this wouldn't be necessary and `include` would work.  That's one argument for supporting `$data_dir` parameter, simplifies the base use case.

Reply:
The config hash could default to having data-dir populated.

Reply:
I thought about this more and hard to validate `$config['data-dir']` and the service simply work work without valid data-dir so made it a parameter and that supports doing `include etcd` because we can provide a sane default. The value for path of directory is still `$_config['data-dir']` which means that if someone provides it in config hash  it will take precedence but we still provide a sane default via parameter since it's the one config option that is required to function.

Reply:
But then if someone does `$config_hash = { foo = bar }` or something they will break their install because suddenly data-dir isn't specified.  I think that makes this more fragile and more easily allows a user to break their install.  If we make `$data_dir` parameter we guarantee a valid value and less prone to user error.  I think only way I'd be comfortable keeping data-dir exclusively in $config is if we make $config use a custom datatype where we define data-dir as a required value and also have to list all possible values for the config hash which makes things less flexible but only way to ensure data-dir is always defined.

Reply:
To some degree we have to protect users from themselves, it would be very confusing if someone says `$config => {... }` and suddenly etcd stops working because they didn't include our default value.

Reply:
Also with the change I made a few minutes ago this still works:

```
$config => {
 'data-dir' => '/some/path',
 ...other options..
}
```

The value for `$config` will bypass our default parameter. But it would also still work if they do this:

```
$config => {
...options not including data-dir...
}
```

I feel like this includes best of both worlds, if a user defines `data-dir` the parameter is ignored but if users don't, they are still going to get a working install thanks to default from parameter.

Reply:
You should always be able to `include class` and have it work without specifying any extra parameters. I also don't like the two levels of parameters where some are actual params and others are hidden in the config hash. Let's put this in the config hash and set the default.

```
Hash $config = { 'data-dir' => '/var/lib/etcd', }
```

And verify that it is there. This way etcd will not break, it just will not compile.

Reply:
Change made